### PR TITLE
[DEVTOOLS-361] Fix python version cache issue

### PR DESCRIPTION
**Pull request by rodrigorbl**
Recently, many github actions is failing with `bad interpreter: No such file or directory` error. According to this
https://github.com/actions/setup-python/issues/182#issuecomment-1151091964, the problem is caused by using a `major.minor` python version as key for caching steps. The `actions/setup-python@v4` step can install different version of python (3.8.13 or 3.8.14, for example) what breaks the cache.

Other issue is that `python` could not be installed on the container used by `python` workers to execute the workflow. Installing python on each step should fix the problem.

### CODE REVIEW

File: .github/actions/prepare_fixit_venv.yml [hunk 1]
```diff
@@ -36,18 +36,18 @@ runs:
...
       run: |
         set -x
         curl -LJO https://raw.githubusercontent.com/carta/.github/main/fixit-requirements.txt
         md5=($(md5sum fixit-requirements.txt))  # the outer parentheses removes the file name
-        echo "::set-output name=venv_hash::fixit-venv-${{ inputs.PYTHON_VERSION }}-$md5-${{ steps.latest_version.outputs.latest_version }}-${{ inputs.FIXIT_LINTER_VERSION }}-v0.1.1"
-    - uses: actions/setup-python@v4
-      id: python-version-setup
-      with:
-        python-version: ${{ inputs.PYTHON_VERSION }}
+        echo "::set-output name=venv_hash::fixit-venv-${{ steps.python-version-setup.outputs.python-version }}-$md5-${{ steps.latest_version.outputs.latest_version }}-${{ inputs.FIXIT_LINTER_VERSION }}-v0.1.1"
```
Comment (rodrigorbl) [inline .github/actions/prepare_fixit_venv.yml:50]:
```suggestion
        echo "::set-output name=venv_hash::fixit-venv-${{ steps.python-version-setup.outputs.version }}-$md5-${{ steps.latest_version.outputs.latest_version }}-${{ inputs.FIXIT_LINTER_VERSION }}-v0.1.1"
```

File: .github/actions/prepare_fixit_venv.yml [hunk 2]
```diff
@@ -36,18 +36,18 @@ runs:
...
       run: |
         set -x
         curl -LJO https://raw.githubusercontent.com/carta/.github/main/fixit-requirements.txt
         md5=($(md5sum fixit-requirements.txt))  # the outer parentheses removes the file name
-        echo "::set-output name=venv_hash::fixit-venv-${{ inputs.PYTHON_VERSION }}-$md5-${{ steps.latest_version.outputs.latest_version }}-${{ inputs.FIXIT_LINTER_VERSION }}-v0.1.1"
-    - uses: actions/setup-python@v4
-      id: python-version-setup
-      with:
-        python-version: ${{ inputs.PYTHON_VERSION }}
+        echo "::set-output name=venv_hash::fixit-venv-${{ steps.python-version-setup.outputs.version }}-$md5-${{ steps.latest_version.outputs.latest_version }}-${{ inputs.FIXIT_LINTER_VERSION }}-v0.1.1"
```
Comment (rodrigorbl) [inline .github/actions/prepare_fixit_venv.yml:50]:
```suggestion
        echo "::set-output name=venv_hash::fixit-venv-${{ steps.python-version-setup.outputs.python-version }}-$md5-${{ steps.latest_version.outputs.latest_version }}-${{ inputs.FIXIT_LINTER_VERSION }}-v0.1.1"
```

**Review by jimmylai — APPROVED** [review]

File: .github/actions/prepare_fixit_venv.yml (hunk 1 above)
Comment (jimmylai) [inline .github/actions/prepare_fixit_venv.yml:50]: It should be outputs.version.
```suggestion
        echo "::set-output name=venv_hash::fixit-venv-${{ steps.python-version-setup.outputs.version }}-$md5-${{ steps.latest_version.outputs.latest_version }}-${{ inputs.FIXIT_LINTER_VERSION }}-v0.1.1"
```

File: .github/actions/prepare_fixit_venv.yml [hunk 3]
```diff
@@ -36,18 +36,19 @@ runs:
...
+      id: python-version-setup
+      with:
+        python-version: ${{ inputs.PYTHON_VERSION }}
     - name: download fixit-requirements.txt
       id: fixit_venv_hash
       shell: bash
       run: |
         set -x
         curl -LJO https://raw.githubusercontent.com/carta/.github/main/fixit-requirements.txt
+        echo "${{ steps.python-version-setup.outputs.python-version }}"
```
Comment (rodrigorbl) [inline .github/actions/prepare_fixit_venv.yml:49]:
```suggestion
        echo "PYTHON VERSION=${{ steps.python-version-setup.outputs.python-version }}"
```

File: .github/actions/prepare_fixit_venv.yml (hunk 1 above)
Comment (rodrigorbl) [reply .github/actions/prepare_fixit_venv.yml:51]: I think this was renamed, based on the documentation: https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#outputs-and-environment-variables

**Review by EricVogl-carta — APPROVED** [review]
Thanks, I didn't realize bumping python in carta-web would break this

**Review by jimmylai — APPROVED** [review]
LGTM

**Pull request merged**